packages/idos-sdk-js/src/lib/enclave.ts

```diff
@@ -95,4 +95,9 @@ export class Enclave {
 
     return this.provider.backupPasswordOrSecret(callbackFn);
   }
+
+  async discoverUserEncryptionKey() {
+    const result = await this.provider.discoverUserEncryptionKey();
+    return result;
```

Comment:
Unless you want to force resolution, I'd not `await` here:
```suggestion
    return this.provider.discoverUserEncryptionKey();
```